### ISSUE TITLE
SEC-1309: Namespace configurations should support Spring EL

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-security#1553]
[AngerClown](https://jira.spring.io/secure/ViewProfile.jspa?name=angerclown@yahoo.com) (Migrated from [SEC-1309](https://jira.spring.io/browse/SEC-1309?redirect=false)) said:

The security namespace is very useful for default configurations, but I would also like to be able to abstract it a little more and do something like the following:

 <http>
    <intercept-url pattern="/**" access="ROLE_USER" />

```
<intercept-url pattern="#{ systemProperties['image.dir'] + '/**' }" filters="none" />
<intercept-url pattern="#{ systemProperties['css.dir'] + '/**' }" filters="none" />
<intercept-url pattern="#{ systemProperties['js.dir'] + '/**' }" filters="none" />

<intercept-url pattern="#{ '/' + systemProperties['login.page'] }" access="IS_AUTHENTICATED_ANONYMOUSLY" requires-channel="https" />
<form-login login-page="#{ '/' + systemProperties['login.page'] }" />

<port-mappings>
  <port-mapping http=#{ systemProperties['http.port'] }"  https=#{ systemProperties['https.port'] }" /> </port-mappings>

<logout />
```

  </http>

In other words, I would like to be able to configure some of this through properties referenced via Spring EL.

Right now this doesn't work at all and I am forced to hand configure the required beans. This is not a major issue since I can get this to work, but long term, it makes maintenance harder since someone has to understand more about the underlying mechanics to be able to create the correct bean definitions.
